pkg/apply/applier.go

```diff
@@ -64,26 +64,6 @@ func (a *Applier) prepareObjects(localInv inventory.Info, localObjs object.Unstr
 	for _, localObj := range localObjs {
 		inventory.AddInventoryIDAnnotation(localObj, localInv)
 	}
-	// If the inventory uses the Name strategy and an inventory ID is provided,
-	// verify that the existing inventory object (if there is one) has an ID
-	// label that matches.
-	// TODO(seans): This inventory id validation should happen in destroy and status.
-	if localInv.Strategy() == inventory.NameStrategy && localInv.ID() != "" {
```

Comment:
was this validation moved somewhere?

Reply:
Does the new code support both NameStrategy and LabelStrategy?

Reply:
This was moved to `getClusterInventoryObjsByName`. It's only needed for NameStrategy because LabelStrategy already filters by matching ID/label